From 7b7f3ce926f4a5c466d72e625fe93102db8ad685 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Wed, 28 May 2014 15:39:31 +0300
Subject: [PATCH 174/213] Version 0.9.5: Solved huge ArrayOutOfBoundsException
 in DataHandler

Solved a huge bug causing an ArrayOutOfBoundsException in DataHandler
Really lucky this did not go onto production ;)
---
 .../org/cgiar/ilri/mistro/farmer/backend/DataHandler.java    | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
index d462bb8..b1df021 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
@@ -665,9 +665,9 @@ public class DataHandler
                         String[][] sireRes = databaseHelper.runSelectQuery(readableDB, databaseHelper.TABLE_COW, columns, selection, null, null, null, null, null);
                         if(sireRes.length == 1){
                             Sire sire = new Sire();
-                            sire.setName(sireRes[cowIndex][1]);
-                            sire.setEarTagNumber(sireRes[cowIndex][2]);
-                            sire.setDateAdded(sireRes[cowIndex][9]);
+                            sire.setName(sireRes[0][1]);
+                            sire.setEarTagNumber(sireRes[0][2]);
+                            sire.setDateAdded(sireRes[0][9]);
 
                             currCow.setSire(sire);
                         }
@@ -685,9 +685,9 @@ public class DataHandler
                         String[][] damRes = databaseHelper.runSelectQuery(readableDB, databaseHelper.TABLE_COW, columns, selection, null, null, null, null, null);
                         if(damRes.length == 1){
                             Dam dam = new Dam();
-                            dam.setName(damRes[cowIndex][1]);
-                            dam.setEarTagNumber(damRes[cowIndex][2]);
-                            dam.setDateAdded(damRes[cowIndex][9]);
+                            dam.setName(damRes[0][1]);
+                            dam.setEarTagNumber(damRes[0][2]);
+                            dam.setDateAdded(damRes[0][9]);
 
                             currCow.setDam(dam);
                         }
-- 
2.7.1

